src/App.tsx

```diff
@@ -1,96 +1,50 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
+import { Select } from './components/Select';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+const items = getNumbers(1, 42).map(n => `Item ${n}`);
+
+function itemsPerPage(perPageNum: number) {
+  const newItems = [...items];
+  const resultItems = [];
+
+  for (let i = 0; i < newItems.length; i += perPageNum) {
```

Comment:
You are using a 'for' loop to create a new array. It's better to use 'map' method in this case.